src/EditorFeatures/CSharpTest/Squiggles/ErrorSquiggleProducerTests.cs

```diff
@@ -122,27 +124,54 @@ void Test()
                     }
                 };
 
+            var diagnosticsAndSpans = await _producer.GetDiagnosticsAndErrorSpans(workspace, analyzerMap);
+
             var spans =
-                (await _producer.GetDiagnosticsAndErrorSpans(workspace, analyzerMap)).Item2
-                    .OrderBy(s => s.Span.Span.Start).ToImmutableArray();
+                diagnosticsAndSpans.Item1
+                    .Zip(diagnosticsAndSpans.Item2, (diagnostic, span) => (diagnostic, span))
+                    .OrderBy(s => s.span.Span.Span.Start).ToImmutableArray();
 
             Assert.Equal(3, spans.Length);
-            var first = spans[0];
-            var second = spans[1];
-            var third = spans[2];
+            var first = spans[0].span;
+            var second = spans[1].span;
+            var third = spans[2].span;
+
+            var expectedToolTip = new ContainerElement(
+                ContainerElementStyle.Wrapped,
+                new ClassifiedTextElement(
+                    new ClassifiedTextRun(ClassificationTypeNames.Text, "IDE0005"),
+                    new ClassifiedTextRun(ClassificationTypeNames.Punctuation, ":"),
+                    new ClassifiedTextRun(ClassificationTypeNames.WhiteSpace, " "),
+                    new ClassifiedTextRun(ClassificationTypeNames.Text, CSharpAnalyzersResources.Using_directive_is_unnecessary)));
 
             Assert.Equal(PredefinedErrorTypeNames.Suggestion, first.Tag.ErrorType);
-            Assert.Equal(CSharpAnalyzersResources.Using_directive_is_unnecessary, first.Tag.ToolTipContent);
+            ToolTipAssert.EqualContent(expectedToolTip, first.Tag.ToolTipContent);
             Assert.Equal(40, first.Span.Start);
             Assert.Equal(25, first.Span.Length);
 
+            expectedToolTip = new ContainerElement(
+                ContainerElementStyle.Wrapped,
+                new ClassifiedTextElement(
+                    new ClassifiedTextRun(ClassificationTypeNames.Text, "IDE0005"),
+                    new ClassifiedTextRun(ClassificationTypeNames.Punctuation, ":"),
+                    new ClassifiedTextRun(ClassificationTypeNames.WhiteSpace, " "),
+                    new ClassifiedTextRun(ClassificationTypeNames.Text, CSharpAnalyzersResources.Using_directive_is_unnecessary)));
+
             Assert.Equal(PredefinedErrorTypeNames.Suggestion, second.Tag.ErrorType);
-            Assert.Equal(CSharpAnalyzersResources.Using_directive_is_unnecessary, second.Tag.ToolTipContent);
+            ToolTipAssert.EqualContent(expectedToolTip, second.Tag.ToolTipContent);
             Assert.Equal(82, second.Span.Start);
             Assert.Equal(60, second.Span.Length);
 
+            expectedToolTip = new ContainerElement(
+                ContainerElementStyle.Wrapped,
+                new ClassifiedTextElement(
+                    new ClassifiedTextRun(ClassificationTypeNames.Text, "IDE0049"),
```

Comment:
Are we able to assert that one of these has a link somewhere?

Reply:
It is possible to implement this such that the link can be tested.

Reply:
Links are now tested